src/Public/New-FunctionFromType.ps1

```diff
@@ -0,0 +1,153 @@
+    If your module uses a prefix (like Dba, Sql, Az, etc), use this Prefix.
+
+    .PARAMETER Passthru
+    Output code to console instead of writing to disk.
+
+    .PARAMETER ExcludeProperty
+    Exclude properties you don't want.
+
+    .EXAMPLE
+    PS> New-FunctionFromType -TypeName Bogus.Data
+
+    Generates 4 new files: New-Data, Get-Data, Set-Data, Remove-Data.
+
+    .EXAMPLE
+    PS> New-FunctionFromType -TypeName Bogus.Data -Verb New
+
+    Generates one new file: New-Data with ConfirImpact of low.
+
+    .EXAMPLE
+    PS> New-FunctionFromType -TypeName Microsoft.SqlServer.Management.Smo.Mail.MailProfile -Prefix Dba -ExcludeProperty UserData -Path C:\temp\mail -ConfirmImpact High -Template dbatools
+
+    Generates 4 new files: New-DbaMailProfile, Get-DbaMailProfile, Set-DbaMailProfile, Remove-DbaMailProfile in C:\temp\mail. Excludes UserData and Parent.
+
+ #>
+
+    [cmdletbinding(DefaultParameterSetName = 'TypeName')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    param (
+        [ValidateSet('All', 'New', 'Remove', 'Get', 'Set')]
+        [String[]]$Verb = "All",
+        [String]$Path = $PWD,
+        [Parameter(Mandatory, ValueFromPipeline, ParameterSetName = 'TypeName')]
+        [String]$TypeName,
+        [Parameter(Mandatory, ValueFromPipeline, ParameterSetName = 'Object')]
+        [Object]$InputObject,
+        [String]$Template = "Default",
+        [ValidateSet("Low", "Medium", "High")]
+        [String]$ConfirmImpact = "Low",
+        [String]$Prefix,
+        [Switch]$Passthru,
+        [String[]]$ExcludeProperty
+    )
+
+    begin {
+        if ($Verb -eq "All") {
+            $Verb = "New", "Remove", "Get", "Set"
+        }
+
+        if ($Template -eq "Default") {
+            $Template = Resolve-Path -Path "$script:PSModuleRoot\Private\template.txt"
+        }
+        elseif ($Template -eq "dbatools") {
+            $Template = Resolve-Path -Path "$script:PSModuleRoot\Private\template-dbatools.txt"
+        }
+        else {
+            if (-not (Test-Path -Path $Template)) {
+                throw "Can't find $Template"
+            }
+        }
+        if ($Path -and -not (Test-Path -Path $Path)) {
+            $null = New-Item -ItemType Directory -Path $Path
+        }
+    }
+    process {
+        if ($PSCmdlet.ParameterSetName -ne 'TypeName') {
+            $TypeName = $InputObject.GetType().Name
+        }
+
+        $properties = @(
+            ($TypeName -as [Type]).GetProperties() | Where-Object { $_.CanWrite }
+            ($TypeName -as [Type]).GetFields()     | Where-Object { -not $_.Attributes.HasFlag([System.Reflection.FieldAttributes]::InitOnly) }
+        )
+
+        if ($ExcludeProperty) {
+            $properties = $properties | Where-Object Name -notin $ExcludeProperty
+        }
+
+        foreach ($verbname in $Verb) {
+            $text = Get-Content -Path $Template
+            $shortname = ($TypeName -split "\.")[-1]
```

Comment:
This should probably be `$shortTypeName` to differentiate from the other names that are being used in here.

Reply:
makes sense, done